### DISCUSSION TITLE
Integrate Ghostty with Fish shell and Zellij (keybindings, auto launching)

### DISCUSSION THREAD

**Discussion by romanzipp** [ghostty-org/ghostty#3740] [Q&A] [answered]
I've tinkered around with Ghostty and was really impressed of how well it followed standards on macOS. So while switching from WezTerm to Ghostty, I've also brought my favorite tools [Fish Shell](https://fishshell.com/) (alternative to bash/zsh)  and [Zellij](https://zellij.dev/) (alternative to tmux).

I will resolve this discussion myself in the following comment so people looking for the same stack of Ghostty + Fish + Zellij will have a nice starting point - hope this is fine!

### Requirements

- Being able to use `CMD + [n, t, w]` and `CMD + OPTION + [left, right]` to navigate tabs in Zellij
- Being able to use `CMD + C` to copy highlighted shell contents
- Auto launching Zellij when opening a Ghostty Terminal window

**Comment by romanzipp** [comment, marked as answer]
I wrote about my motivations and some other config tweaks [in a blog post](https://romanzipp.com/blog/ghostty-zellij-fish-shell), the solution below is a TLDR for this discussion.

## `ghostty` Config file

See [Ghostty Configuration docs](https://ghostty.org/docs/config), I've placed my config in `~/.config/ghostty/ghostty`.

```
keybind = cmd+t=unbind
keybind = cmd+n=unbind
keybind = cmd+c=unbind
keybind = cmd+w=unbind
keybind = cmd+opt+left=unbind
keybind = cmd+opt+right=unbind
```

### Fish `config.fish` Config file

See the [Fish tutorial docs](https://fishshell.com/docs/current/tutorial.html) for more info, typically the config should be placed in `~/.config/fish/config.fish`.

```fish
# Unset the default fish greeting text which messes up Zellij
set fish_greeting

# Check if we're in an interactive shell
if status is-interactive

    # At this point, specify the Zellij config dir, so we can launch it manually if we want to
    export ZELLIJ_CONFIG_DIR=$HOME/.config/zellij

    # Check if our Terminal emulator is Ghostty
    if [ "$TERM" = "xterm-ghostty" ]
        # Launch zellij
        eval (zellij setup --generate-auto-start fish | string collect)
    end
end
```

### Zellij `config.kdl` Config file

Above, we've specified the `ZELLIJ_CONFIG_DIR`, so place your `config.kdl` in that folder.

```kdl
keybinds {
    normal {
        bind "Super c" { Copy; }
        bind "Super Alt Left" { GoToPreviousTab; }
        bind "Super Alt Right" { GoToNextTab; }
        bind "Super w" { CloseTab; }
        bind "Super n" { NewTab; }
    }
}

# If Zellij is not starting into fish shell
default_shell "fish"

// macOS specific
copy_command "pbcopy"
copy_on_select false
```

**Reply by dbritto-dev** [reply]
@romanzipp everything works for me except that zellij does not run fish at start so I need to type fish, do you know how to fix that?

**Reply by romanzipp** [reply]
Try to add `default_shell "fish"` to your Zellij config

**Reply by dbritto-dev** [reply]
@romanzipp `default_shell "fish"` works but without context, when I try to run `node --version` I got `fish: Unknown command: node` in the other hand if type `fish` manually everything works as expected

**Reply by Stef16Robbe** [reply]
@dbritto-dev I have not tried this specific config, but I have ran into this before. I'm pretty sure this is because when you start a user session in a different shell and then open fish, it takes over your environment (variables) including your $PATH from that existing session. fish has a `--login` option, maybe try that?

**Reply by dbritto-dev** [reply]
@Stef16Robbe doesn't work

**Comment by yuukikuno** [comment]
I'll leave this here in case it helps someone: if you are encountering issues with `shell-integration` features (like the somewhat recent `ssh-terminfo` and `ssh-env` options from https://github.com/ghostty-org/ghostty/pull/7608), you might want to `source "$GHOSTTY_RESOURCES_DIR"/shell-integration/fish/vendor_conf.d/ghostty-shell-integration.fish` as per https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup. (See also https://github.com/ghostty-org/ghostty/discussions/6062#discussioncomment-12377170)
